share/shwap/namespaced_data.go

```diff
@@ -0,0 +1,205 @@
+package shwap
+
+import (
+	"crypto/sha256"
+	"fmt"
+
+	"github.com/celestiaorg/celestia-app/pkg/wrapper"
+	"github.com/celestiaorg/nmt"
+	nmt_pb "github.com/celestiaorg/nmt/pb"
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+	pb "github.com/celestiaorg/celestia-node/share/shwap/proto"
+)
+
+// NamespacedData stores collections of RowNamespaceData, each representing shares and their proofs
+// within a namespace.
+type NamespacedData []RowNamespaceData
+
+// Flatten combines all shares from all rows within the namespace into a single slice.
+func (ns NamespacedData) Flatten() []share.Share {
+	var shares []share.Share
+	for _, row := range ns {
+		shares = append(shares, row.Shares...)
+	}
+	return shares
+}
+
+// RowNamespaceData holds shares and their corresponding proof for a single row within a namespace.
+type RowNamespaceData struct {
+	Shares []share.Share `json:"shares"` // Shares within the namespace.
+	Proof  *nmt.Proof    `json:"proof"`  // Proof of the shares' inclusion in the namespace.
+}
```

Comment:
RowIndex can be derived from RowID

Reply:
How will the `GetSharesByNamesapce` will look like then?

Reply:
The way it is right now. One idea we had with Vlad is to break the getter to take shwap IDs as an input, but we havent decided on it

Reply:
The way it is right now means that it will require a header and namespace as input. How can I get RowIndex in this case?

Reply:
I see your question now. We can provide a helper func that taken the DAH and namespace gives you the idx.

Reply:
Or that can be a method on type returned frok GetSharesByNamespace, but it should not be part of the core type and transferree over the wire

Reply:
Ok, thanks. It would also be nice if this `method` could calculate the index of the share in the EDS, so we can get rid of this logic inside blob module(can work on it actually). It will simplify the code base and will keep all share-related stuff together.